tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp

```diff
@@ -0,0 +1,373 @@
+}
+
+struct PatternCallback {
+    const char* name;
+    CGPatternDrawPatternCallback callback;
+};
+
+struct StencilColor {
+    const char* name;
+    CGFloat color[4];
+};
+
+PatternCallback stencilPatternCallback[] = { { "drawStencilStar", &drawStencilStar }, { "drawStencilBox", &drawStencilBox } };
+
+StencilColor stencilColors[] = { { "red", { 1, 0, 0, 1 } }, { "teal", { 0, 0.5, 0.5, 1 } }, { "purple", { 0.5, 0, 0.5, 1 } } };
+
+int patternsizes[] = { 16, 20, 32 };
+
+//<PatternCallback,StencilColor,pattern size>
+class CGPatternStencil : public WhiteBackgroundTest<>,
+                         public ::testing::WithParamInterface<::testing::tuple<PatternCallback, StencilColor, int>> {
+    CFStringRef CreateOutputFilename() {
+        PatternCallback stencil = ::testing::get<0>(GetParam());
+        StencilColor stencilColor = ::testing::get<1>(GetParam());
+        int size = ::testing::get<2>(GetParam());
+
+        return CFStringCreateWithFormat(nullptr,
+                                        nullptr,
+                                        CFSTR("TestImage.CGContext.CGPatternStencil.%s.%s.%d.png"),
+                                        stencil.name,
+                                        stencilColor.name,
+                                        size);
+    }
+};
+
+TEST_P(CGPatternStencil, DrawStencils) {
+    CGPatternDrawPatternCallback callback = ::testing::get<0>(GetParam()).callback;
+    const CGFloat* stencilColor = ::testing::get<1>(GetParam()).color;
+    int size = ::testing::get<2>(GetParam());
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    static const CGPatternCallbacks callbacks = { 0, callback, NULL };
+
+    CGColorSpaceRef baseSpace = CGColorSpaceCreateDeviceRGB();
+    CGColorSpaceRef patternSpace = CGColorSpaceCreatePattern(baseSpace);
+    CGContextSetFillColorSpace(context, patternSpace);
+    CGColorSpaceRelease(patternSpace);
+    CGColorSpaceRelease(baseSpace);
+
+    CGPatternRef pattern = CGPatternCreate(
+        NULL, CGRectMake(0, 0, size, size), CGAffineTransformIdentity, size, size, kCGPatternTilingConstantSpacing, false, &callbacks);
+
+    CGContextSetFillPattern(context, pattern, stencilColor);
+    CGPatternRelease(pattern);
+    CGContextFillRect(context, bounds);
+}
+
+INSTANTIATE_TEST_CASE_P(CGPatternTests,
+                        CGPatternStencil,
+                        ::testing::Combine(::testing::ValuesIn(stencilPatternCallback),
+                                           ::testing::ValuesIn(stencilColors),
+                                           ::testing::ValuesIn(patternsizes)));
+
+#pragma endregion Stencil Pattern
+
+#pragma region Colored Pattern
+
+void drawColoredPatternMultiSquare(void* info, CGContextRef context) {
+    CGFloat subunit = 5; // the pattern cell itself is 16 by 18
+
+    CGRect myRect1 = { { 0, 0 }, { subunit, subunit } }, myRect2 = { { subunit, subunit }, { subunit, subunit } },
```

Comment:
nit: this is not one of those cases where multiple variables on the same line makes sense 🌮